lib/fed_email_domain_downloader.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+require 'csv'
+require 'faraday'
+require 'pry'
+require 'fileutils'
+
+class FedEmailDomainDownloader
+  attr_reader :destination
+
+  DOT_GOV_DOWNLOAD_PATH = 'https://raw.githubusercontent.com/cisagov/dotgov-data/main/current-federal.csv'
+
+  def initialize(destination: 'tmp/fed_download_path')
+    @destination = destination
+  end
+
+  def dot_gov_csv_path
+    response = Faraday.get(DOT_GOV_DOWNLOAD_PATH)
+    response.body
+  end
+
+  def run!
+    FileUtils.mkdir_p(destination)
+    csv ||= CSV.parse(dot_gov_csv_path, col_sep: ',', headers: true)
+    File.open("#{destination}/fed_email_domains.txt", 'w') do |file|
+      csv.each do |row|
+        file.write("#{row['Domain name']}\n")
+      end
+    end
+  end
+end
+
+FedEmailDomainDownloader.new.run!
```

Comment:
I think it's generally a bad practice to run side effects at the top level of a file like this, I'd recommend wrapping it like below. That way, if it's explicitly called as a command line script, it will run, otherwise, it will be loaded into memory and can be called explicitly, such as in a `deploy/activate` kind of situation

```suggestion
if $PROGRAM_NAME == __FILE__
  FedEmailDomainDownloader.new.run!
end
```